### DISCUSSION TITLE
Player isActive behaviour

### DISCUSSION THREAD

**Discussion by MikeIsotech** [pschatzmann/arduino-audio-tools#1552] [Q&A] [answered]
I'm developing software based on an a player using audio tools for WAV files using SDMMC and I2S.

I need to know when the active file has finished playing. The player is configured to not do auto next. I found that player.isActive() returned true even when the track had finished.  A bit of checking in AudioPlayer.h moveToNextFileOnTimeout()  function shows that the procedure returns without clearing active if not using auto next. Further down in the function it there is a section of code that can't be reached, because of the early return, would clear active flag. Commenting out the first two lines of the function ensures that things work as expected.

Have I discovered a bug? Apologies if this is not the correct way to report this.

**Comment by pschatzmann** [comment, marked as answer]
Just check how many bytes the copy() command was processing. If it is 0 you are at the end!
By the way this is also explained in the Copy section of the [The most important basics!](https://github.com/pschatzmann/arduino-audio-tools/wiki/Introduction). I suggest that you spend some time reading the Wiki
